ddtrace/internal/_encoding.pyx

```diff
@@ -213,10 +215,12 @@ cdef class MsgpackEncoderBase(BufferedEncoder):
         ret = msgpack_pack_array(&self.pk, L)
         if ret != 0: raise RuntimeError("Couldn't pack trace")
 
-        if L > 0 and trace[0].context is not None:
-            dd_origin = trace[0].context.dd_origin
-        else:
-            dd_origin = None
+        if L > 0 and trace[0].context is not None and trace[0].context.dd_origin is not None:
+            IF PY_MAJOR_VERSION >= 3:
+                dd_origin_bytestring = trace[0].context.dd_origin.encode("UTF-8")
+                dd_origin = dd_origin_bytestring
+            ELSE:
+                dd_origin = trace[0].context.dd_origin
```

Comment:
I _think_ you can just do:

```
dd_origin = <char*> trace[0].context.dd_origin
```

and then Cython handles the rest. you can compile with `cythonize(..., annotate=True)` to see what it generates, iirc it will do the coercion for you.

Reply:
we checked, no you cannot. handling of `bytes` is not handled by casting alone.

@Yun-Kim can you add a comment here to explain that?

Reply:
Yes! I am actually going to pull those dd_origin improvement changes into a separate PR.